leakcanary-object-watcher-android/src/main/java/leakcanary/internal/DefaultCanaryLog.kt

```diff
@@ -30,10 +25,10 @@ internal class DefaultCanaryLog : Logger {
     message: String,
     vararg args: Any?
   ) {
-    d(
-        String.format(message, *args) + '\n'.toString() + Log.getStackTraceString(
-            throwable
-        )
-    )
+    d("${String.format(message, args)}\n${Log.getStackTraceString(throwable)}")
```

Comment:
Are you 100% that we don't need the spread operator here? String.format takes an object, and args is an object, so I'm worried it's going to run toString on the args array itself, hence the spread operator

Reply:
Should be the same method yes. I've definitely had this kind of issues in java in the past so I'm worried :)

Reply:
You were absolutely right! Even though it navigates to the same method it doesn't behave in the same way and results in an exception:
```
@Test fun loggerTest() {
  d("Too many placeholders %s %s %s", "one", "two", "three")
}
fun d(message: String, vararg args: Any?) {
  val formatted = String.format(message, args)
  println(formatted)
}

java.util.MissingFormatArgumentException: Format specifier '%s'
	at java.util.Formatter.format(Formatter.java:2519)
	at java.util.Formatter.format(Formatter.java:2455)
	at java.lang.String.format(String.java:2940)
```

I looked into the decompiled code, and here's what it's doing (omitted unrelated code):
```
public final void d(@NotNull String message, @NotNull Object... args)

//when using String.format(message, args)
Object[] var5 = new Object[]{args};
String formatted = String.format(message, Arrays.copyOf(var5, var5.length));
System.out.println(formatted);

//when using String.format(message, *args)
Object[] var5 = Arrays.copyOf(args, args.length);
String formatted = String.format(message, Arrays.copyOf(var5, var5.length));
System.out.println(formatted);
```
`Object[] var5 = new Object[]{args};` this resulted in creating another array with array inside of it:

![image](https://user-images.githubusercontent.com/2152942/63559718-da55a680-c507-11e9-8e88-90aa43cd824b.png)
****

**This raises two questions:**

1. Should I just revert things related to spread operator back and disable the check? It might cause more hidden problems like this one. Furthermore, if `SharkLog` gets a lambda parameter, the spread operator will not be needed.
2. Is it a bug in the Kotlin plugin or somewhere that it points to the same method but ends up to be different things? I feel like it should be raised as an issue, what do you think?

Reply:
1. => How about we revert / disable the check for now, and then follow up with the inline sharklog?
2. Probably worth filing. There might be some cases where this makes sense, but at the very least there should be more education on this.

Reply:
1. Will do.

2. https://discuss.kotlinlang.org/t/report-missing-spread-operator-for-vararg/14003